File: javascript/Anagram Detection.js
// write the function isAnagram
var isAnagram = function(test, original) {
  
  let t = test.toLowerCase()
  let o = original.toLowerCase()
  
  if(t.length !== o.length) return false
  
  let map = {}
  
  for(let char of t){
    map[char] = (map[char] || 0) + 1 
  }
  
  for( let char of o){
    if(!map[char]) return false
    map[char]--
  }
  
  return true
};

File: python/Anagram Detection.py
# write the function is_anagram
def is_anagram(test, original):
    
    t = test.lower()
    o = original.lower()
    
    if len(t) != len(o):
        return False
    
    seen = {}
    
    for char in t:
        seen[char] = seen.get(char,0) +1
        
    for char in o:
        if char not in seen or seen[char] == 0:
            return False
        seen[char] -= 1
        
    return True